internal/parser/parser_test.go

```diff
@@ -2286,38 +2286,263 @@ k / l < m + n - o or p() <= q() or r >= s and not t =~ /a/ and u !~ /a/`,
 											Name:     "s",
 										},
 									},
-								},
-								Right: &ast.UnaryExpression{
-									BaseNode: base("2:47", "2:59"),
-									Operator: ast.NotOperator,
-									Argument: &ast.BinaryExpression{
-										BaseNode: base("2:51", "2:59"),
-										Operator: ast.RegexpMatchOperator,
-										Left: &ast.Identifier{
-											BaseNode: base("2:51", "2:52"),
-											Name:     "t",
-										},
-										Right: &ast.RegexpLiteral{
-											BaseNode: base("2:56", "2:59"),
-											Value:    regexp.MustCompile(`a`),
+									Right: &ast.UnaryExpression{
+										BaseNode: base("2:47", "2:59"),
+										Operator: ast.NotOperator,
+										Argument: &ast.BinaryExpression{
+											BaseNode: base("2:51", "2:59"),
+											Operator: ast.RegexpMatchOperator,
+											Left: &ast.Identifier{
+												BaseNode: base("2:51", "2:52"),
+												Name:     "t",
+											},
+											Right: &ast.RegexpLiteral{
+												BaseNode: base("2:56", "2:59"),
+												Value:    regexp.MustCompile("a"),
+											},
 										},
 									},
 								},
+								Right: &ast.BinaryExpression{
+									BaseNode: base("2:64", "2:72"),
+									Operator: ast.NotRegexpMatchOperator,
+									Left: &ast.Identifier{
+										BaseNode: base("2:64", "2:65"),
+										Name:     "u",
+									},
+									Right: &ast.RegexpLiteral{
+										BaseNode: base("2:69", "2:72"),
+										Value:    regexp.MustCompile("a"),
+									},
+								},
 							},
-							Right: &ast.BinaryExpression{
-								BaseNode: base("2:64", "2:72"),
-								Operator: ast.NotRegexpMatchOperator,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "logical operators precedence",
+			raw: `not a or b
+a or not b
+not a and b
+a and not b
+a and b or c
+a or b and c
+
+not (a or b)
+not (a and b)`,
```

Comment:
If I put more parenthesized stuff it fails:

```
not a or b
a or not b
not a and b
a and not b
a and b or c
a or b and c

not (a or b)
not (a and b)
(a or b) and c
a and (b or c)
```

It seemed like it merged lines 9 and 10

Reply:
Each of those should be a separate test. It is a bit weird that it merged the lines though since they should be separate statements.

Reply:
Yep it is so. Do you prefer I add a test for that hat gets skipped?

I'll separate tests

Reply:
I don't think that's needed. Can you file a separate bug for it though so it can be investigated and fixed?

Reply:
I filed the bug: https://github.com/influxdata/flux/issues/1102

Anyhow I put and skipped the test in the last commit, just to ease reproduction and avoid regression in the future.